Documentation/HPC-Packaging.md

```diff
@@ -33,84 +33,146 @@ To set up the build of a specific compiler/MPI combo the following steps
 are required (anything in [..] is optional):
 
 1. create an entry to the `_multibuild` file of the form:
-```
-  <package>$compiler_family[${compiler_family_version}]-[${mpi_family}[${mpi_family_version}]]-]hpc[-${other_modifier}]</package>
-```
-  like:
-```
-  <multibuild>
-  ...
-    <package>gnu6-openmpi1-hpc</package>
-  ...
-  </multibuild>
-```
+
+   ```
+   <package>
+   $compiler_family[${compiler_family_version}]-[${mpi_family}[${mpi_family_version}]]-]hpc[-${other_modifier}]
+   </package>
+   ```
+like:
+    ```
+    <multibuild>
+    ...
+    <package>gnu-openmpi3-hpc</package>
+    ...
+    </multibuild>
+   ```
 
 2. In the spec file add a line:
-```
-   %global flavor @BUILD_FLAVOR@
-```
+    ```
+    %global flavor @BUILD_FLAVOR@%{nil}
+    ```
+
+The macro `%{nil}` will avoid that the syntax of the spec file becomes
+incorrect if `@BUILD_FLAVOR@` expands to nothing.
 
 3. Define:
-    1. Macro `%pname`:
+    1. Macro `%pname` with the package name:
     ```
     %define pname <package_name>
     ```
-    2. Macro `%vers` (the package version):
+    2. Macro `%vers` with the package version:
     ```
     %define vers <package_version>
     ```
-    3. Macro `%_vers` (the package version with underscores):
+    3. Macro `%_vers` with the package version with underscores:
     ```
-    %define %_vers <package_version_with-_-replacing-.>
+    %define _vers %(echo %{vers} | tr . _)
```

Comment:
This may not work under all circumstances: if _vers is required by the prebuild parser which (among other things) checks for dependencies. 
So you may want to add a note that this may have to be specified explicitly.